addons/website_slides/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,94 @@
+        for user, partner_id in user_partner_id_lst:
+            delta_karma = earned_karma - init_earned_karma_by_partner_id.get(partner_id, 0)
+            user.karma = user_karma[user] + delta_karma
+
+    @api.model
+    def _merge_partners_for_unique_m2m(self, src_partners, dst_partner,
+                                       res_model, other_record_field, order=None, prefer_dst_partner=False):
+        """ Merge records of res_model for the given partners following the given preferences.
+
+        ! It is the caller responsibility to check the ACL, this method is executed in sudo.
+        Use this method when the model ("join" model that links uniquely partner to another
+        model) contains a unique key per partner (foreign key to the other model), to select
+        the records to assign to the dst_partner from the record of src_partners and dst_partner.
+        Indeed, when there is an uniq constraint on partner_id and another field we cannot simply
+        assign all record of src_partners to dst_partner without potentially create conflicts
+        (unique constraint violation).
+        When multiple records with the same unique key per partner among the record of the
+        given partners (src_partners and dst_partner) are encountered, this method choose
+        the one with the highest preference and assign it to dst_partner. Of course, if there
+        is only one, it is just assigned to the dst_partner.
+
+        :param recordset src_partners: source partners
+        :param recordset dst_partner: destination partner
+        :param str res_model: many-to-many relationship model linking res_partner and another
+            model through the other_record_field
+        :param str other_record_field: field name of the unique "other" record per partner
+        :param str order: order by clause to express the preference (ex.: completed DESC, vote DESC)
+        :param bool prefer_dst_partner: when a conflict occurs whether to always prefer the record
+        with dst_partner if present instead of following the preferences
+        """
+        Model_sudo = self.env[res_model].sudo()
+        records_to_delete = list()
+        grouped_records = groupby(Model_sudo.search_read([('partner_id', 'in', (src_partners + dst_partner).ids)],
+                                                         [other_record_field, 'partner_id'],
+                                                         order=other_record_field + (f', {order}' if order else '')),
+                                  key=lambda r: r[other_record_field][0])
+        if prefer_dst_partner:
+            for _, records in grouped_records:
+                dst_partner_records, others = partition(lambda r: r['partner_id'][0] == dst_partner.id, records)
+                if dst_partner_records:
+                    records_to_delete += others
+                    records_to_delete += dst_partner_records[1:]
```

Comment:
about this line, are you sure ?
```suggestion
```

we already remove `others`, do we need to remove `dst_partner_records[1:]` as well ?
(the test is not crashing if I comment this line)


same question about
```py
records_to_delete += records[1:]
```

if you group  partner with a other partner that has zero slide.partner, it will still remove some